src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1085,7 +1085,10 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         // keybindings on the keyUp, then we'll still send the keydown to the
         // connected terminal application, and something like ctrl+shift+T will
         // emit a ^T to the pipe.
-        if (!modifiers.IsAltGrPressed() && keyDown && _TryHandleKeyBinding(vkey, scanCode, modifiers))
+        if (!modifiers.IsAltGrPressed() &&
+            keyDown &&
+            (_core.TryMarkModeKeybinding(vkey, modifiers) ||
+             _TryHandleKeyBinding(vkey, scanCode, modifiers)))

```

Comment:
What about `if (_core.IsInReadOnlyMode())`?
Shouldn't we call `_core.TryMarkModeKeybinding(vkey, modifiers)` inside `_TryHandleKeyBinding`?